### DISCUSSION TITLE
IAFD Agent - Some Actor Headshots Missing (Red X) while other actors populate correctly in Agent Match

### DISCUSSION THREAD

**Discussion by ms52538** [CodyBerenson/PGMA-Modernized#304] [General]
This may have been discussed before.
My IAFD scene agent is matching correctly on scenes, and on their website the actors have headshot images. When the agent matches an OnlyFans title released under RawFuckClub.com, it's scraping and populating all the data, including actors, but in some cases an actor's headshot appears with a red X and nothing is populated for that actor, even though other actors as part of the scrape are being fully populated. Case in point any scenes starring Felipe Hervera.

With the recent conversation taking place around actors and actor databases, I thought it might be as simple as pulling down a copy of the actor's headshot file and placing it under the _PGMA\Cast\Face subdirectory. I've tried using a few copies, naming them Felipe#Hervera.jpg, Felipe Hervera.jpg, and the actual name of the file as named by IAFD. But no dice. Is it even possible to go out and find a picture of an actor that can be used in all scenes in which he stars in and matches to IAFD without having to create an NFO for each file?  My good buddy John, here, would just populate the pictures into the files themselves with his set of tools. My problem is I use the same files as active torrents through BiglyBT, so I have to change file names at the torrent level but cannot alter the files themselves. Hence I have to rely on a solution that scrapes the data via an agent. Is this matching issue a bug with IAFD? Thanks, as always to the SuperHuman Coding Team Members (eye candy included) ;)

**Comment by JPH71** [comment]
Unfortunately, when scraping at the moment, I have not as yet found a way
to use locally stored images as headshots....
You can create an NFO file and within the file put an image link for the
actor...
Plex should go through the agents and then populate accordingly...
Cody usually uses the NFO Importer and will be able to provide you more
help when it comes to this.
I will be enhancing the agents soon to try and sort a number of issues out.

Kr

Jason

**Reply by ms52538** [reply]
Thank you Jason. I guess I am curious to why I can see a picture of (in this case) both the models in IAFD, yet only one actor's picture is pulled over and populated, the other's is not, and the rest of the scrape fully populates. That actor only has one name (no aliases), and no matter which scene I try to match with him in it, his picture does not populate. This is a broader issue as it is happening with many actors pictures in IAFD. I'm guessing IAFD is the issue, and not the agent?

Again, much appreciated!

**Comment by JPH71** [comment]
Now that is strange...
If the picture is in IAFD it should be pulled...
However, due to cloudflare protections in IAFD, sometimes this fails..
Try rescraping the film...

**Reply by ms52538** [reply]
I have, just now, and multiple times before, on several actors having the same problem; their picture is on IAFD as being part of the scene, but only a RED X appears in Plex for the actor, where others do populate. In some scenes, multiple actors appear and others do not. It seems to be consistent with the actor image file.

**Comment by jfreiman** [comment]
I'm having the same issue, except instad of a RED X.
I'm seeing not images (grey blank circle) with ? under them.  

Before finding this thread, I tried changing the Agent settings, moving them around, etc.

I've been seeing this issue for several days now.  I updated my PGMA to the most recent build (updated a week ago) and I'm still having the same issue -- as expected? Is this something that we will now be dealing with - no cast photos?

Since I am able to load the IAFD pages locally on my computer, is there anyway that the process of scraping the cast photos could be done locally?

**Comment by ms52538** [comment]
I can confirm what John is alluding to as it is also happening for me. Yesterday I worked on several files and scraped them primarily with IADF, WayBig, Fagalicious, etc. for adult scenes and GEVI for adult feature films.

After successfully scraping one scene for example, I checked and everything populated include actor names (but no pictures). I then scaped another scene with one of the actors and this time it did populate his picture, I then went back and checked the previous scene and his picture appeared there too. An hour later, scape more scenes with the same actor, his picture did not render and it disappeared from the other scenes I just mentioned. I checked them again just now, and their picture is populating in the scenes mentioned. However I look at other scenes that successfully scraped yesterday, and actor pictures are missing, and underneath the actor name is a ? question mark. i.e. below, both Trevor Harris and DeAngelo Jackson's pictures are not appearing, but Jayden Marcos' is.

Trevor Harris, Jayden Marcos and DeAngelo Jackson in "The Freaks Next Door"
2024    58min    X
[⚣ Bareback](http://127.0.0.1:32400/web/index.html#!/media/d1419a89d6e1cafd7b8ed0c62a8fd04bbddedb36/com.plexapp.plugins.library?source=136&key=%2Flibrary%2Fsections%2F136%2Fall%3Ftype%3D1%26genre%3D301586&pageType=list),[⚣ Group Sex (3+)](http://127.0.0.1:32400/web/index.html#!/media/d1419a89d6e1cafd7b8ed0c62a8fd04bbddedb36/com.plexapp.plugins.library?source=136&key=%2Flibrary%2Fsections%2F136%2Fall%3Ftype%3D1%26genre%3D301583&pageType=list),[⚣ Interracial](http://127.0.0.1:32400/web/index.html#!/media/d1419a89d6e1cafd7b8ed0c62a8fd04bbddedb36/com.plexapp.plugins.library?source=136&key=%2Flibrary%2Fsections%2F136%2Fall%3Ftype%3D1%26genre%3D302337&pageType=list)
⪢ IAFD Film ❔ ⪢ IAFD Cast ✅ Yes / ❌ No ⪢ IAFD Error 403 ❔Film, Cast ⪢ Fagalicious


[Cast & Crew](http://127.0.0.1:32400/web/index.html#!/server/d1419a89d6e1cafd7b8ed0c62a8fd04bbddedb36/credits?key=%2Flibrary%2Fmetadata%2F1250923&metadataType=movie&type=actor)
[DeAngelo Jackson](http://127.0.0.1:32400/web/index.html#!/media/d1419a89d6e1cafd7b8ed0c62a8fd04bbddedb36/com.plexapp.plugins.library?source=136&key=%2Flibrary%2Fsections%2F136%2Fall%3Factor%3D516064&pageType=list)
❔

[Jayden Marcos](http://127.0.0.1:32400/web/index.html#!/media/d1419a89d6e1cafd7b8ed0c62a8fd04bbddedb36/com.plexapp.plugins.library?source=136&key=%2Flibrary%2Fsections%2F136%2Fall%3Factor%3D514808&pageType=list)
✅
[Trevor Harris](http://127.0.0.1:32400/web/index.html#!/media/d1419a89d6e1cafd7b8ed0c62a8fd04bbddedb36/com.plexapp.plugins.library?source=136&key=%2Flibrary%2Fsections%2F136%2Fall%3Factor%3D516065&pageType=list)
❔
Fagalicious Collection

**Comment by fourstix** [comment]
I'm getting the same 403 (Forbidden) error.  It looks like Cloudflare is blocking the requests.

**Comment by CodyBerenson** [comment]
Since this is a proper issue, issue #306 has been opened.  Please contribute any further dialog there.  Thanks.
